In [8]:
import pandas as pd
import numpy as np

In [9]:
df = pd.read_csv("titanic_data.csv")
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
884         0       2    male  27.0      0      0  13.0000        S  Second   
885         1       1  female  19.0      0      0  30.0000        S   First   
886         0       3  female   NaN      1      2  23.4500        S   Third   
887         1       1    male  26.0      0      0  30.0000        C   First   
888         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
884    man        True  NaN  Southampton    no   True  
885  woman       False    B  Southampton   yes   True  
886  woman       False  NaN  Southampton    no  False  
887    man        True    C    Cherbourg   yes   True  
888    man        True  NaN   Queenstown    no   True  

[889 rows x 15 columns]

In [10]:
df.shape

(889, 15)

In [11]:
df.info

<bound method DataFrame.info of      survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
884         0       2    male  27.0      0      0  13.0000        S  Second   
885         1       1  female  19.0      0      0  30.0000        S   First   
886         0       3  female   NaN      1      2  23.4500        S   Third   
887         1       1    male  26.0      0      0  30.0000        C   First   
888         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_m

In [13]:
df.dropna(subset=['deck'], inplace=True)

In [14]:
df.shape

(203, 15)

In [15]:
df.info

<bound method DataFrame.info of      survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
1           1       1  female  38.0      1      0  71.2833        C  First   
3           1       1  female  35.0      1      0  53.1000        S  First   
6           0       1    male  54.0      0      0  51.8625        S  First   
10          1       3  female   4.0      1      1  16.7000        S  Third   
11          1       1  female  58.0      0      0  26.5500        S  First   
..        ...     ...     ...   ...    ...    ...      ...      ...    ...   
869         1       1  female  47.0      1      1  52.5542        S  First   
870         0       1    male  33.0      0      0   5.0000        S  First   
877         1       1  female  56.0      0      1  83.1583        C  First   
885         1       1  female  19.0      0      0  30.0000        S  First   
887         1       1    male  26.0      0      0  30.0000        C  First   

       who  adult_male deck  em

In [16]:
df.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town',
       'alive', 'alone'],
      dtype='object')

In [26]:
df.groupby(['sex'])['survived'].agg('sum')

sex
female    91
male      45
Name: survived, dtype: int64

136 in total survived

In [49]:
df['fare'].mean()

76.34170837438424

The average fare is 76.34

In [63]:
df.groupby(['fare'])

In [72]:
df.groupby(['sex'])['fare'].mean()

sex
female    86.817268
male      66.755583
Name: fare, dtype: float64

The average fare for males is 66.76
The average fare for females is 86.82

In [73]:
df.groupby(['sex', 'age'])['fare'].mean()

sex     age 
female  2.0      81.006250
        4.0      27.850000
        14.0    120.000000
        15.0    211.337500
        16.0     61.293067
                   ...    
male    64.0    263.000000
        65.0     44.264600
        70.0     71.000000
        71.0     34.654200
        80.0     30.000000
Name: fare, Length: 96, dtype: float64

Above is the average fare and age of females and males from each town.